Makefile

```diff
@@ -75,38 +165,80 @@ starkgate-contracts-legacy:
 	cp ./lib/starkgate-contracts-old/starkgate-artifacts/starkware/starknet/std_contracts/upgradability_proxy/proxy.json ./artifacts/proxy_starkgate.json
 	cp ./lib/starkgate-contracts-old/starkgate-artifacts/starkware/starknet/std_contracts/ERC20/ERC20.json ./artifacts/ERC20.json
 
-braavos-account-cairo:
+# Target: braavos-account-cairo
+# Builds Braavos account contracts
+# Prerequisites: ensure-asdf
+braavos-account-cairo: ensure-asdf
 	# Building
-	asdf install scarb 2.8.4 && asdf global scarb 2.8.4
-	cd ./lib/braavos-account-cairo && \
- 	git checkout $(BRAAVOS_CONTRACTS_COMMIT_HASH) && \
- 	~/.asdf/installs/scarb/2.8.4/bin/scarb build
+	@if [ "$$(uname)" = "Darwin" ] && command -v brew >/dev/null 2>&1; then \
+		. "$$(brew --prefix asdf)/libexec/asdf.sh" && \
+		cd ./lib/braavos-account-cairo && \
+		git checkout $(BRAAVOS_CONTRACTS_COMMIT_HASH) && \
+		. "$$(brew --prefix asdf)/libexec/asdf.sh" && \
+		asdf install scarb 2.8.4 && \
+		. "$$(brew --prefix asdf)/libexec/asdf.sh" && \
+		asdf local scarb 2.8.4 && \
+		scarb build; \
+	else \
+		. "$(HOME_DIR)/.asdf/asdf.sh" && \
+		cd ./lib/braavos-account-cairo && \
+		git checkout $(BRAAVOS_CONTRACTS_COMMIT_HASH) && \
+		. "$(HOME_DIR)/.asdf/asdf.sh" && \
+		asdf install scarb 2.8.4 && \
+		. "$(HOME_DIR)/.asdf/asdf.sh" && \
+		asdf local scarb 2.8.4 && \
+		scarb build; \
```

Comment:
can this part be moved outside because it's common

Reply:
done